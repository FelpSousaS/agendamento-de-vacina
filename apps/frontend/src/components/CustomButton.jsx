import React from 'react';
import { Button } from '@chakra-ui/react';
import buttonStyles from '../style/buttonStyles';

const CustomButton = ({ variant = 'primary', children, ...props }) => {
  return (
    <Button {...props} {...buttonStyles[variant]}>
      {children}
    </Button>
  );
};

export default CustomButton;
