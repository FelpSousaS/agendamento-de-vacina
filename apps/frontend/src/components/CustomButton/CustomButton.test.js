import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import CustomButton from './CustomButton';
import renderer from 'react-test-renderer';

describe('<CustomButton />', () => {
  test('renders button with provided text', () => {
    render(<CustomButton variant="primary">Primary Button</CustomButton>);
    expect(screen.getByText('Primary Button')).toBeInTheDocument();
  });

  test('matches snapshot for primary variant', () => {
    const tree = renderer
      .create(<CustomButton variant="primary">Primary Button</CustomButton>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('matches snapshot for secondary variant', () => {
    const tree = renderer
      .create(<CustomButton variant="secondary">Secondary Button</CustomButton>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('applies additional props to the button', () => {
    const { container } = render(
      <CustomButton variant="primary" aria-label="Test Button">
        Test
      </CustomButton>,
    );
    const button = container.querySelector('button');

    if (button) {
      expect(button).toHaveAttribute('aria-label', 'Test Button');
    } else {
      throw new Error('Button element not found');
    }
  });

  test('calls onClick handler when clicked', () => {
    const handleClick = jest.fn();
    const { container } = render(
      <CustomButton variant="primary" onClick={handleClick}>
        Click Me
      </CustomButton>,
    );
    const button = container.querySelector('button');

    if (button) {
      fireEvent.click(button);
      expect(handleClick).toHaveBeenCalledTimes(1);
    } else {
      throw new Error('Button element not found');
    }
  });
});
