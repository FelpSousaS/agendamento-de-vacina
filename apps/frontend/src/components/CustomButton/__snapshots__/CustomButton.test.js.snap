// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomButton /> matches snapshot for primary variant 1`] = `
<button
  className="chakra-button css-sns2y8"
  type="button"
>
  Primary Button
</button>
`;

exports[`<CustomButton /> matches snapshot for secondary variant 1`] = `
<button
  className="chakra-button css-lqx85q"
  type="button"
>
  Secondary Button
</button>
`;
